akka-actor/src/main/scala/akka/util/ConstantFun.scala

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2015-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+
+import akka.annotation.InternalApi
+import akka.japi.function.{ Function ⇒ JFun, Function2 ⇒ JFun2 }
+import akka.japi.{ Pair ⇒ JPair }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object ConstantFun {
```

Comment:
hmm, indeed, Akka HTTP use it for some historical reasons. I think here we can just move `ConstantFun` without worrying about Akka HTTP using it because it uses a specific version of Akka. The end users  upgrading Akka but not Akka HTTP will break nothing. The only thing we should care is that when upgrading Akka for Akka HTTP, we should change the usage of `ConstantFun`, or maybe use its own copy in Akka HTTP project.